### PR TITLE
Update server connection validation auto-feature for mpOpenAPI 4

### LINKED ISSUE #28918: Update server connection validation integration with MP OpenAPI to support 4.0 feature
The projects `io.openliberty.rest.handler.config.openapi.2.0` and `io.openliberty.rest.handler.validator.openapi.2.0` use the MP OpenAPI feature to provide OpenAPI documentation for the REST API for [validating server connections](https://openliberty.io/docs/latest/validating-server-connections.html).

We need to make this functionality work with `mpOpenAPI-4.0`.

- [ ] New/updated implementation bundles
- [ ] New/updated autofeature
- [ ] FAT test repeats updated

### PR DESCRIPTION
**Pull request by benjamin-confino**
- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

Resolves #28918

### CODE REVIEW

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#build

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.configValidationConfigSchema-4.0.feature
Comment (Azquelt) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.configValidationConfigSchema-4.0.feature:1]: I think we can just extend the IBM-ProvisionCapability of `configValidationConfigSchema-3.0` to cover mpOpenAPI-4.0 as well.

We have separate versions for mpConfig 1.x, 2.x and 3.x because they contain different content, but there's no reason to have a new one for 4.x if the content is the same.

Same for `configValidationValidatorSchema-4.0`

File: dev/com.ibm.ws.rest.handler.config_fat/publish/servers/com.ibm.ws.rest.handler.config.openapi.fat/jvm.options [hunk 1]
```diff
@@ -0,0 +1 @@
+-Dcom.ibm.ws.beta.edition=true
```
Comment (Azquelt) [inline dev/com.ibm.ws.rest.handler.config_fat/publish/servers/com.ibm.ws.rest.handler.config.openapi.fat/jvm.options]: I don't think this is necessary unless we have beta guards in code. The FAT tests run with all features available anyway.

**Review by Azquelt — COMMENTED** [review]
I've left a couple of small comments but otherwise this looks good.

**Comment by benjamin-confino** [comment]
#build

File: dev/io.openliberty.microprofile.openapi.4.0.internal.services/bnd.bnd [hunk 2]
```diff
@@ -35,17 +35,18 @@ Private-Package: io.openliberty.microprofile.openapi40.internal.services.impl
...
+	io.openliberty.io.smallrye.jandex3;version=latest,\
+	io.openliberty.microprofile.openapi.2.0.internal;version=latest,\
+	io.openliberty.org.eclipse.microprofile.openapi.4.0,\
+	com.ibm.ws.adaptable.module;version=latest,\
+	com.ibm.ws.logging;version=latest,\
+	com.ibm.websphere.org.osgi.core;version=latest,\
+	com.ibm.websphere.org.osgi.service.component;version=latest,\
+	com.ibm.wsspi.org.osgi.service.component.annotations;version=latest,\
+	com.ibm.ws.org.osgi.annotation.versioning;version=latest,\
+	com.ibm.ws.kernel.boot.core
```
Comment (Azquelt) [inline dev/io.openliberty.microprofile.openapi.4.0.internal.services/bnd.bnd]: I'm unsure why this change was needed, but I don't see it causing any harm.

**Review by Azquelt — DISMISSED** [review]

File: dev/io.openliberty.microprofile.openapi.4.0.internal.services/bnd.bnd (hunk 2 above)
Comment (benjamin-confino) [reply dev/io.openliberty.microprofile.openapi.4.0.internal.services/bnd.bnd]: Oh I know what happened. That was for the other PR to `import com.ibm.ws.kernel.productinfo.ProductInfo;` and entered the wrong commit.

**Comment by benjamin-confino** [comment]
#build

**Review by Azquelt — APPROVED** [review]

**Comment by Azquelt** [comment]
#libby

**Comment by benjamin-confino** [comment]
The pipeline was a mess with stuff like Caused by: java.io.FileNotFoundException: [https://libfsfe06.hursley.ibm.com//liberty/personal/2/benjamin-confino-29485-20240828-1054-_w0YcIWUiEe-O2sUHUib4kg/fe/../autoFVT_bucket_zips/com.ibm.ws.jpa.tests.spec10.packaging_jpa_3.0_fat.zip](https://libfsfe06.hursley.ibm.com//liberty/personal/2/benjamin-confino-29485-20240828-1054-_w0YcIWUiEe-O2sUHUib4kg/autoFVT_bucket_zips/com.ibm.ws.jpa.tests.spec10.packaging_jpa_3.0_fat.zip)

Which won't be related to the autofeatures, the build was good though so going by that, this is safe to merge
